File: portal_manager_hub_ui/lib/models/container_info.dart
class ContainerInfo {
  final String id, image, names, command, status, ports, createdAt;

  ContainerInfo({
    required this.id,
    required this.image,
    required this.names,
    required this.command,
    required this.status,
    required this.ports,
    required this.createdAt,
  });

  factory ContainerInfo.fromJson(Map<String, dynamic> json) => ContainerInfo(
    id: json['ID'] ?? '',
    image: json['Image'] ?? '',
    names: json['Names'] ?? '',
    command: json['Command'] ?? '',
    status: json['Status'] ?? '',
    ports: json['Ports'] ?? '',
    createdAt: json['CreatedAt'] ?? '',
  );
}

File: container-core-api-manager/src/main/java/com/deploymanager/dto/ContainerInfo.java
package com.deploymanager.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

/**
 * Representa un contenedor Docker según salida JSON de `docker ps --format '{{json .}}'`
 */
@Data
public class ContainerInfo {

    @JsonProperty("ID")
    private String id;

    @JsonProperty("Image")
    private String image;

    @JsonProperty("Names")
    private String names;

    @JsonProperty("Command")
    private String command;

    @JsonProperty("CreatedAt")
    private String createdAt;

    @JsonProperty("Status")
    private String status;

    @JsonProperty("Ports")
    private String ports;

}
